acmg_db/views/search_views.py

```diff
@@ -0,0 +1,133 @@
+from ..models import *
+from ..forms import SearchForm
+from ..utils.variant_utils import get_variant_hash
+from django.contrib.auth.decorators import login_required
+from django.db import transaction
+from django.shortcuts import render, redirect, get_object_or_404
+from django.http import HttpResponseNotFound
+import re
+
+@transaction.atomic
+@login_required
+def search(request):
+	"""
+	Page to allow user to search for samples, genes and variants.
+	"""
+
+	# make empty form
+	form = SearchForm()
+
+	message = None
+
+	# if form submitted
+	if request.method == 'POST':
+		form = SearchForm(request.POST)
+
+		if form.is_valid():
+			
+			search_input = form.cleaned_data['search_input'].upper().strip()
+
+			# check if we've searched for sample
+			x = re.search("^\d{2}M\d{5}", search_input)
```

Comment:
There are a few samples in the live database at the moment that this wont pick up, but they're all free text so it'll be really hard to catch them all (e.g. NA or GENQA). 
These samples can still be searched in the 'Reporting' tab though, so we should get them to use that if they need to view these samples